### PR TITLE
DRA-1222: test id setup

### PR DESCRIPTION
**Pull request by jesperlauridsen**
First stab at data-testid's on _every_ interactive element.

I think i got them all. I decided to make the function have 4 args, as follows, to create the id:

type - which kinda interaction we're talking about.
component - which component this is included in, first line of seperation.
title - more differentation based on the title of the interative element in question
index - the number in the line.

These 4 should make it fairly easy to make sure we don't have colliding test elements, imho.

Added to all i could find. If im missing any, we can add them - or if our testers find any, then we can just add them on the run.

Have a look at it, give it a spin and see if you're happy with it.

### CODE REVIEW

File: src/components/common/wc-spot-item.ts [hunk 1]
```diff
@@ -36,10 +37,24 @@ class SpotComponent extends HTMLElement {
 	connectedCallback() {
 		//TODO: We're waiting for real data here
 		if (this.data) {
+			console.log(this.data);
```
Comment (jorntx) [inline src/components/common/wc-spot-item.ts:40]: Lets get rid of this and remove those in TimeSearchFilters.vue as well :)

**Review by jorntx — COMMENTED** [review]
This looks good and I tried to challenge the id cleanup method a bit but it passed. Only some cleanup 

I am slightly worried about performance as the app grows - all those test-ids are actually reactive as it is now but mitigating this is a bit cumbersome so lets leave it be for now.

File: src/components/common/wc-spot-item.ts [hunk 2]
```diff
@@ -36,10 +37,24 @@ class SpotComponent extends HTMLElement {
...
+				link.setAttribute('data-testid', addTestDataEnrichment('link', 'spot-component', `link-${this.data.id}`, 0));
+			/* link?.addEventListener('click', (event) => {
+				console.log('clicked!');
+				event.preventDefault();
+				window.dispatchEvent(
+					new CustomEvent('change-path', {
+						detail: { path: this.data ? this.data.id : '' },
+					}),
+				);
+			}); */
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/common/wc-spot-item.ts:57]: No, not right now. We're looking into a full rework of the spot-component i think - might even just convert it to a vue component now - and that piece of code might be handy.

**Review by jorntx — APPROVED** [review]
As agreed we will go with this  for now

**Comment by jorntx** [comment]
@jesperlauridsen can you resolve merge conflicts?

**Pull request merged**